metricflow-semantics/tests_metricflow_semantics/model/semantics/test_linkable_element_set.py

```diff
@@ -202,6 +202,46 @@ def _linkable_set_with_uniques_and_duplicates() -> LinkableElementSet:
     )
 
 
+def test_linkable_elements_for_path_key() -> None:
+    """Tests accessing the linkable element tuples for a given path key.
+
+    The keys all share the same name and links but should return different results. Note the metric keys have
+    additional entity link annotations due to the way we differentiate between link paths within the query and link
+    paths outside the query (from outer query to inner metric query).
+    """
+    linkable_set = _linkable_set_with_uniques_and_duplicates()
+    entity_key = ElementPathKey(
+        element_name=AMBIGUOUS_NAME, element_type=LinkableElementType.ENTITY, entity_links=(_base_entity_reference,)
+    )
+    dimension_key = ElementPathKey(
+        element_name=AMBIGUOUS_NAME, element_type=LinkableElementType.DIMENSION, entity_links=(_base_entity_reference,)
+    )
+    ambiguous_metric_key = ElementPathKey(
+        element_name=AMBIGUOUS_NAME,
+        element_type=LinkableElementType.METRIC,
+        entity_links=(_base_entity_reference,),
+        metric_subquery_entity_links=(_base_entity_reference, _base_entity_reference),
+    )
+    doubled_ambiguous_metric_key = ElementPathKey(
+        element_name=AMBIGUOUS_NAME,
+        element_type=LinkableElementType.METRIC,
+        entity_links=(_base_entity_reference, _base_entity_reference),
+        metric_subquery_entity_links=(_base_entity_reference, _base_entity_reference),
+    )
```

Comment:
@courtneyholcomb is this expected? The doubling of `_base_entity_reference` in the `metric_subquery_entity_links` seems a bit odd to me, and even more so in the `entity_links` section. Is this an artifact of how I constructed the join paths, or is this expected in the general case?

Reply:
No, there shouldn't be a duplicate entity in the same path (either path). Where did you see that returned? I can look into it

Reply:
@courtneyholcomb - it's in this test. The definitions of these objects are above, should be lines 114-167 for the metric objects.